ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLtextures.cpp

```diff
@@ -81,9 +81,19 @@ unsigned char* graphics_copy_texture_pixels(int texture, unsigned* fullwidth, un
 
 unsigned char* graphics_copy_texture_pixels(int texture, int x, int y, int width, int height) {
   unsigned fw, fh;
-  //We could use glCopyImageSubData here, but it's GL4.3
+  //We could use glCopyImageSubData here instead of cropping, but it's GL4.3
```

Comment:
This is identical to the GL3 method. I thought we had a common "OpenGL" folder for when this happens.

Reply:
Alright: 985c977713a2560891905c9561c96284c4bd8e80

That's good, that shows me you're still paying attention. I wasn't sure of the interface yet, hence why I hadn't deduplicated this sources yet. In truth, it's ok now for us to totally deduplicate them to the `OpenGL/` folder. Neither does anything different than the other. The part that used to, samplers, was moved with the state handling.

Also, I threw out the garbage we were using to check for mipmapping or anisotropy support. We're using GLEW, which already has defines for us to check.
https://github.com/enigma-dev/enigma-dev/pull/1722/commits/985c977713a2560891905c9561c96284c4bd8e80#diff-a2d95b5539b5d2acd0b1d984333009e6R157

I also deduplicated TheExDeus's custom texture create function by just having the mandatory one call his. I changed the signature on it though to accept the correct GL types since it seems he didn't finish the user-space enum he was planning.
https://github.com/enigma-dev/enigma-dev/pull/1722/commits/985c977713a2560891905c9561c96284c4bd8e80#diff-a2d95b5539b5d2acd0b1d984333009e6R64

Please watch out, and make sure nobody adds something to this file to turn it into GL3 only and break GL1.1 users. Yes, I also went through and checked all of the texture functions we are calling, they are all GL1.1. However, I do need to point out that GL_BGRA textures are actually GL1.2. So that means any game that uses texturing/sprites/backgrounds in ENIGMA is actually GL1.2 minimum at the moment.